### PR TITLE
feat: AI Fix for IntelliJ ⚡️ [IDE-580]

### PR DESCRIPTION
**Pull request by cat2608**
### Description

Make issue data available for the clients rendering AI Fixes and update UI.

This change need https://github.com/snyk/snyk-intellij-plugin/pull/596

#### Tickets:

- IDE-579
- IDE-580

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

### CODE REVIEW

File: infrastructure/code/code_html_test.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func Test_Code_Html_getCodeDetailsHtml(t *testing.T) {
 	}
 
 	// invoke method under test
-	codePanelHtml := getCodeDetailsHtml(issue)
+	codePanelHtml := getCodeDetailsHtml(issue, "")
```
Comment (ShawkyZ) [inline infrastructure/code/code_html_test.go:55]: add a folderPath here and check the generated response

File: infrastructure/code/template/details.html [hunk 2]
```diff
@@ -808,8 +819,8 @@ <h2 class="ai-fix-header">
               <p>⚡ Fix this issue by generating a solution using Snyk DeepCode AI</p>
 
               <div class="sn-fix-wrapper">
-                <button id="generate-ai-fix" class="generate-ai-fix">✨ Generate fix <span class="wide">using Snyk
-                    DeepCode AI</span></button>
+                <button id="generate-ai-fix" folder-path="{{.FolderPath}}" file-path="{{.FilePath}}"
```
[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply infrastructure/code/template/details.html:822]: Nah, VSCode still uses it's local rendering. BUT, we should add a chore to activate the server supplied html/css in VSCode

**Review by bastiandoetsch — APPROVED** [review]
Looks good! Please consider updating the tests as Shawky suggested.

File: infrastructure/code/code_html_test.go [hunk 3]
```diff
@@ -152,7 +152,7 @@ func Test_Code_Html_getCodeDetailsHtml_ignored(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (acke) [reply infrastructure/code/code_html_test.go:158]: Fixed!

File: infrastructure/code/code_html_test.go (hunk 1 above)
Comment (acke) [reply infrastructure/code/code_html_test.go]: Fixed!

**Review by ShawkyZ — APPROVED** [review]

**Pull request merged**